signac/common/config.py

```diff
@@ -145,6 +147,9 @@ def load_config(root=None, local=False):
     """Load configuration, searching upward from a root path."""
     if root is None:
         root = os.getcwd()
+    if not os.path.isdir(root):
+        with pytest.raises(ValueError):
+            raise ValueError(f"The root directory '{root}' does not exist.")
```

Comment:
The `with pytest.raises` piece should be written in a test for this function, but doesn't need to be used in the implementation.
```suggestion
    if not os.path.isdir(root):
        raise ValueError(f"The root directory '{root}' does not exist.")
```

For the test, you can look at [test_project.py](https://github.com/glotzerlab/signac/blob/master/tests/test_project.py) and add a couple lines above these ones: https://github.com/glotzerlab/signac/blob/4bf9bceacfc780de49a879e16a45382ae0a932c9/tests/test_project.py#L2286-L2287

I would add these lines to the `test_get_project` function:
```python
        with pytest.raises(ValueError):
            signac.get_project(root="/a/path/that/does/not/exist")
```

Reply:
Thank you for the suggestions! I'll add them in. I want to make sure that the error currently being raised is actually as expected.  I'm getting 12 warnings, all in the new ValueError I set up. I'll copy an example below. As far as I can tell, the temporary directory set up for testing is registering as Not A Directory when tested, though in non-tests and non-temp directories the load_config() works as expected. Is it expected for the tempfile.TemproraryDirectory() to behave like this?

`root = '/var/folders/cl/2y21xs_55pdg2wg0rgf130qc0000gn/T/signac_u3cgrvrc/a', local = False

    def load_config(root=None, local=False):
        """Load configuration, searching upward from a root path."""
        if root is None:
            root = os.getcwd()
        if not os.path.isdir(root):
>           raise ValueError(f"The root directory '{root}' does not exist.")
E           ValueError: The root directory '/var/folders/cl/2y21xs_55pdg2wg0rgf130qc0000gn/T/signac_u3cgrvrc/a' does not exist.`

Reply:
I'm not sure if I understand. Can you push your proposed changes to this branch, and I can try to reproduce locally?

Reply:
Yes, thank you! I've uploaded it; I hope it merges? Here's a screenshot of part of the output of "python -m pytest tests/" for extra clarity.

Basically, I think I could make it pass the tests by going through and specifying "raises(ValueError)" for all of these points, but I'm unsure that would actually catch issues as we want it to.
<img width="1305" alt="image" src="https://user-images.githubusercontent.com/57418606/182194938-56a5c03e-c268-4908-acaa-22ccc0202cf3.png">